python/lsst/ap/association/diaPipe.py

```diff
@@ -532,50 +693,93 @@ def run(self,
-            raise RuntimeError(
-                "Duplicate DiaObjects (updated) created after "
-                "DiaCalculation. This is unexpected behavior and should be "
-                "reported. Exiting.")
+    @timeMethod
+    def runForcedMeasurement(self, diaObjects, updatedDiaObjects, exposure, diffIm, idGenerator,
+                             preloadedDiaForcedSources):
+        """Forced Source Measurement
 
-        if self.config.doRunForcedMeasurement:
-            # Force photometer on the Difference and Calibrated exposures using
-            # the new and updated DiaObject locations.
-            diaForcedSources = self.diaForcedSource.run(
-                diaCalResult.diaObjectCat,
-                diaCalResult.updatedDiaObjects.loc[:, "diaObjectId"].to_numpy(),
-                exposure,
-                diffIm,
-                idGenerator=idGenerator)
-        else:
-            # alertPackager needs correct columns
-            diaForcedSources = pd.DataFrame(columns=[
-                "diaForcedSourceId", "diaObjectID", "ccdVisitID", "psfFlux", "psfFluxErr",
-                "ra", "dec", "midpointMjdTai", "band",
-            ])
+        Forced photometry on the difference and calibrated exposures using the
+        new and updated DiaObject locations.
 
-        # Store DiaSources, updated DiaObjects, and DiaForcedSources in the
-        # Apdb.
+        Parameters
+        ----------
+        diaObjects : `pandas.DataFrame`
+            Catalog of DiaObjects.
+        updatedDiaObjects : `pandas.DataFrame`
+            Catalog of updated DiaObjects.
+        exposure : `lsst.afw.image.ExposureF`
+            Calibrated exposure differenced with a template to create
+            ``diffIm``.
+        diffIm : `lsst.afw.image.ExposureF`
+            Difference image exposure in which the sources in ``diaSourceCat``
+            were detected.
+        idGenerator : `lsst.meas.base.IdGenerator`
+            Object that generates source IDs and random number generator seeds.
+        preloadedDiaForcedSources : `pandas.DataFrame`
+            Catalog of previously detected forced DiaSources, from the APDB.
+
+        Returns
+        -------
+        diaForcedSources : `pandas.DataFrame`
+            Catalog of calibrated forced photometered fluxes on both the
+            difference and direct images at DiaObject locations.
+        """
+        # Force photometer on the Difference and Calibrated exposures using
+        # the new and updated DiaObject locations.
+        diaForcedSources = self.diaForcedSource.run(
+            diaObjects,
+            updatedDiaObjects.loc[:, "diaObjectId"].to_numpy(),
+            exposure,
+            diffIm,
+            idGenerator=idGenerator)
+        # columns "ra" and "dec" are required for spatial sharding in Cassandra
+        diaForcedSources.rename(columns={"coord_ra": "ra", "coord_dec": "dec"}, inplace=True)
```

Comment:
These lines were removed on `main` after you made your branch, and should be removed from the refactor.